File: shared/script-types/examples/mightymike-player-sway.ts
import type { MightyMikeLifecycleModule } from "../src/games/mightymike";
import { hasTag, makeHorizontalSwayOffset } from "./helpers";

const PLAYER_TAG = "mightymike.player";
const PLAYER_SWAY_SPEED = 6;
const PLAYER_SWAY_DISTANCE = 6;

export function isMightyMikePlayer(tags: readonly string[]): boolean {
  return hasTag(tags, PLAYER_TAG);
}

export const onObjectFrame: NonNullable<
  MightyMikeLifecycleModule["onObjectFrame"]
> = (ctx) => {
  if (!isMightyMikePlayer(ctx.tags)) {
    return;
  }

  return makeHorizontalSwayOffset(
    ctx.levelTimeSeconds,
    PLAYER_SWAY_SPEED,
    PLAYER_SWAY_DISTANCE,
  );
};

File: shared/script/examples/mightymike-player-sway.lua
local helpers = require("helpers")
local module = {}

local PLAYER_TAG = "mightymike.player"
local PLAYER_SWAY_SPEED = 6
local PLAYER_SWAY_DISTANCE = 6

function module.onObjectFrame(ctx)
    if not helpers.hasTag(ctx.tags, PLAYER_TAG) then
        return nil
    end

    return helpers.makeHorizontalSwayOffset(
        ctx.levelTimeSeconds,
        PLAYER_SWAY_SPEED,
        PLAYER_SWAY_DISTANCE
    )
end

return module
